GliaWidgetsTests/Sources/Coordinators/Chat/ChatCoordinatorTests.swift

```diff
@@ -0,0 +1,400 @@
+    }
+
+    // View model
+
+    func test_correctViewModelForChat() {
+        let viewController = coordinator.start()
+
+        switch viewController.viewModel {
+        case .chat: XCTAssertTrue(true)
+        default: XCTFail()
+        }
+    }
+
+    func test_correctViewModelForSecureTranscript() {
+        let viewController = createCoordinator(startWithSecureTranscriptFlow: true)
+            .start()
+
+        switch viewController.viewModel {
+        case .transcript: XCTAssertTrue(true)
+        default: XCTFail()
+        }
+    }
+
+    // Delegate for chat model
+
+    // Take media is not tested because this triggers the native
+    // "App would like to access the camera" dialog, which could
+    // bring unintended consequences.
+    func test_chatModelPickMedia() throws {
+        let viewController = coordinator.start()
+
+        let scene = try XCTUnwrap(UIApplication.shared.connectedScenes.first as? UIWindowScene)
+        let window = scene.windows.first
+        let oldRootViewController = window?.rootViewController
+        window?.rootViewController = viewController
+        defer { window?.rootViewController = oldRootViewController }
+
+        switch viewController.viewModel {
+        case .chat(let viewModel):
+            viewModel.delegate?(.pickMedia(.init(with: .cancelled)))
+            coordinator.mediaPickerController?.viewController{ shownViewController in
+                XCTAssertNotNil(shownViewController as? UIImagePickerController)
+            }
+        default: XCTFail()
+        }
+    }
+
+    func test_chatModelPickFile() throws {
+        let viewController = coordinator.start()
+
+        let scene = try XCTUnwrap(UIApplication.shared.connectedScenes.first as? UIWindowScene)
+        let window = scene.windows.first
+        let oldRootViewController = window?.rootViewController
+        window?.rootViewController = viewController
+        defer { window?.rootViewController = oldRootViewController }
+
+        switch viewController.viewModel {
+        case .chat(let viewModel):
+            viewModel.delegate?(.pickFile(.init(with: .cancelled)))
+            let filePickerController = coordinator.filePickerController?.viewController as? UIDocumentPickerViewController
+
+            XCTAssertNotNil(filePickerController)
+        default: XCTFail()
+        }
+    }
+
+    func test_chatModelMediaUpgradeAccepted() throws {
+        let viewController = coordinator.start()
+        let mediaOffer: CoreSdkClient.MediaUpgradeOffer = try .init(
+            type: .audio,
+            direction: .twoWay
+        )
+
+        switch viewController.viewModel {
+        case .chat(let viewModel):
+
+            viewModel.delegate?(
+                .mediaUpgradeAccepted(
+                    offer: mediaOffer,
+                    answer: { _, _ in}
+                )
+            )
+        default: XCTFail()
+        }
+
+        coordinator.delegate = { event in
```

Comment:
What if delegate closure is not called for some reason (if it is `nil` for example) or called unexpected number of times? I think these cases will not be caught by this test.
Instead, we can try define array where we are to populate it with events delivered by delegate closure, then we can assert the contents of array. Does it make sense?

Reply:
Makes sense, will change these tests and then will open another PR in which I address this also for the secure conversations tests. Thank you.

Reply:
Fixed df25ce2bc7f83589d7f52bc5992a6ca18c966e3d